distage/distage-framework/src/main/scala/izumi/distage/effect/modules/PolymorphicCatsDIEffectModule.scala

```diff
@@ -0,0 +1,46 @@
+package izumi.distage.effect.modules
+
+import cats.effect.{Async, Bracket, Concurrent, ConcurrentEffect, Sync, Timer}
+import cats.{Applicative, Functor, Monad, MonadError, Parallel}
+import distage.{ModuleDef, TagK}
+import izumi.distage.model.effect.{DIApplicative, DIEffect, DIEffectAsync, DIEffectRunner}
+import izumi.functional.mono.SyncSafe
+import monix.bio.Task
+
+/**
+  * Module definition that binds effect TC instances for an arbitrary F[_].
+  *
+  */
+class PolymorphicCatsDIEffectModule[F[_]: ConcurrentEffect: TagK] extends ModuleDef {
+  implicit private def diEffectRunner: DIEffectRunner[F] =
+    new DIEffectRunner[F] {
+      override def run[A](f: => F[A]): A = ConcurrentEffect[F].toIO(f).unsafeRunSync()
```

Comment:
In this case `Effect[F]` is sufficient (esp. if you extract it into a companion object instance)